backend.go

```diff
@@ -346,6 +346,9 @@ func (b *versionedKVBackend) config(ctx context.Context, s logical.Storage) (*Co
 		if err := proto.Unmarshal(raw.Value, conf); err != nil {
 			return nil, err
 		}
+	} else {
+		// Set config max versions to default
+		conf.MaxVersions = defaultMaxVersions
```

Comment:
Don't do this -- we can not later tell the difference between a user-set value and the default value, in case we ever want to update the default.

Reply:
Makes sense. I changed my fix. Thanks!

Reply:
Sounds good.

(We've learned this the hard way in the past, several times over, so I'm sensitive to it now. It's also why we sometimes store pointers for e.g. bools when we need to know whether the user set it and the zero value is itself meaningful.)